File: sdk/experimental/php/src/Output/Builtins.php
<?php

declare(strict_types=1);

namespace HopTop\Kit\Output;

use HopTop\Kit\Output\Formatter\Builtin\CsvFormatter;
use HopTop\Kit\Output\Formatter\Builtin\JsonFormatter;
use HopTop\Kit\Output\Formatter\Builtin\TableFormatter;
use HopTop\Kit\Output\Formatter\Builtin\TextFormatter;
use HopTop\Kit\Output\Formatter\Builtin\YamlFormatter;

/**
 * Built-in formatter registration: table + json + yaml + csv + text.
 *
 * 'table' is the default --format and intentionally minimal: pipe-friendly
 * ASCII, no borders, no color. Adopters wanting richer tables (borders,
 * alignment, theming) override the 'table' key with their own Formatter.
 *
 * Registry::default() invokes Builtins::register() exactly once.
 */
final class Builtins
{
    public static function register(Registry $r): void
    {
        $r->register(new CsvFormatter());
        $r->register(new JsonFormatter());
        $r->register(new TableFormatter());
        $r->register(new TextFormatter());
        $r->register(new YamlFormatter());
    }
}

File: sdk/experimental/rs/src/output/builtins.rs
//! Built-in formatter registration: table + json + yaml + csv + text.
//!
//! 'table' is the default --format and intentionally minimal: pipe-friendly
//! ASCII via comfy-table's `NOTHING` preset, no borders, no color.
//! Adopters wanting richer tables (borders, UTF-8 box-drawing, theming)
//! register their own Formatter under the 'table' key.

use std::sync::Arc;

use super::registry::Registry;

mod csv;
mod json;
mod rows;
mod table;
mod text;
mod yaml;

pub use csv::CsvFormatter;
pub use json::JsonFormatter;
pub use table::TableFormatter;
pub use text::TextFormatter;
pub use yaml::YamlFormatter;

pub fn register_all(r: &Registry) {
    let _ = r.register(Arc::new(CsvFormatter));
    let _ = r.register(Arc::new(JsonFormatter));
    let _ = r.register(Arc::new(TableFormatter));
    let _ = r.register(Arc::new(TextFormatter));
    let _ = r.register(Arc::new(YamlFormatter));
}
